src/LoginForm.tsx

```diff
@@ -2,89 +2,66 @@ import { useContext, useEffect, useState, ChangeEvent, FormEvent } from "react";
 import { IsLoggedContext } from "./App.tsx";
 
-const LoginForm = () => {
-  const [usersFields, setUsersFields] = useState({ email: "", password: "" });
-  const [error, setError] = useState("");
-  const [isFormSubmitted, setIsFormSubmitted] = useState(false);
-  const { setIsLoggedIn } = useContext(IsLoggedContext);
-
-  const MIN_PASSWORD_LENGTH = 4;
-
-  const onEmailChange = (e: ChangeEvent<HTMLInputElement>): void => {
-    if (e.target !== null) {
-      const inputEl = e.target as HTMLInputElement;
-      setUsersFields({
-        ...usersFields,
-        email: inputEl.value,
-      });
-    }
-  };
-
-  const onPasswordChange = (e: ChangeEvent<HTMLInputElement>): void => {
-    if (e.target !== null) {
-      const inputEl = e.target as HTMLInputElement;
-      setUsersFields({
-        ...usersFields,
-        password: inputEl.value,
-      });
-    }
-  };
-
-  useEffect(() => {
-    if (!isFormSubmitted) {
-      return;
-    }
-    if (usersFields.password.length < MIN_PASSWORD_LENGTH) {
-      setError("Short password");
-      return;
-    } else {
-      setError("");
-    }
-  }, [usersFields, isFormSubmitted]);
+import { SubmitHandler, useForm } from "react-hook-form";
 
-  const onSubmit = (e: FormEvent<HTMLFormElement>) => {
-    const { email, password } = usersFields;
-    e.preventDefault();
+type FormFields = {
+  email: string;
+  password: string;
+};
 
-    setIsFormSubmitted(true);
+const LoginForm = () => {
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors },
+  } = useForm<FormFields>();
+  const { setIsLoggedIn } = useContext(IsLoggedContext);
 
+  const onSubmit: SubmitHandler<FormFields> = ({ email, password }) => {
     const checkUser = usersData.find(
       (user) => user.email === email && user.password === password
     );
 
     if (checkUser) {
       setIsLoggedIn(true);
     } else {
-      setError("Not correct email or password");
+      setError("root", { message: "Not correct email or password" });
     }
   };
 
   return (
     <>
-      <form className="login-form" onSubmit={onSubmit}>
-        <div>
-          <label>Email: </label>
-          <input
-            name="email"
-            type="text"
-            required
-            value={usersFields.email}
-            onChange={onEmailChange}
-          />
-        </div>
-        <div>
-          <label>Password: </label>
-          <input
-            name="password"
-            type="password"
-            required
-            value={usersFields.password}
-            onChange={onPasswordChange}
-          />
-        </div>
+      <form className="login-form" onSubmit={handleSubmit(onSubmit)}>
+        <input
+          {...register("email", {
+            required: "Email is required",
+            validate: (value) => {
+              if (!value.includes("@")) {
+                return "Email must include @";
+              }
+              return true;
+            },
```

Comment:
let's split code as much as possible. 

i would move the following function to `helpers/validators.tsx` due to possibility of re-using

example:
```
const emailValidator = (value:some type): String  {
 if (!value.includes("@")) {
      return "Incorrect format of EMAIL";
  }
}

.......
import {emailValidator} from 'helpers/validators'

{...register("email", {
      required: "Email is required",
      validate: emailValidator
   }}

```